manifests/integrations/directory.pp

```diff
@@ -1,35 +1,72 @@
 # Class: datadog_agent::integrations::directory
 #
-# This class will install the necessary config to hook the directory check
+# This class will install the necessary config to hook the directory in the agent
 #
 # Parameters:
-#   $directory
-#       The directory to gather stats for
-#   $tag_name
-#       The name used to tag the metrics (directory alias)
-#   $pattern
-#       The `fnmatch` pattern to use when reading the "directory"'s files. default "*"
-#   $recursive
-#       Boolean, when true the stats will recurse into directories
+#   directory
+#       (Required) - string, the directory path to monitor
+#       This will be included as a tag: name:<name>.
+#
+#   name
+#       (Optional) - string, tag metrics with specified name. defaults to the "directory"
+#
+#   dirtagname
+#       (Optional) - string, the name of the key for the tag used for the directory, the value will be the value of "name" (see above). The resulting tag will be "<dirtagname>:<name>". defaults to "name"
+#
+#   filetagname
+#       (Optional) - string, the name of the key for the tag used for each file, the value will be the filename. The resulting tag will be "<filetagname>:<filename>". defaults to "filename"
+#
+#   filegauges
+#       (Optional) - boolean, when true stats will be an individual gauge per file (max. 20 files!) and not a histogram of the whole directory. default False
+#
+#   pattern
+#       (Optional) - string, the `fnmatch` pattern to use when reading the "directory"'s files. The pattern will be matched against the files' absolute paths and relative paths in "directory". default "*"
+#
+#   recursive
+#       (Optional) - boolean, when true the stats will recurse into directories. default False
+#
+#   countonly
+#       (Optional) - boolean, when true the stats will only count the number of files matching the pattern. Useful for very large directories.
+#
 #
 # Sample Usage:
 #
-#  class { 'datadog_agent::integrations::directory' :
-#      directory     => '/mnt/media',
-#      tag_name      => 'name',
-#      pattern       => '*',
-#      recursive     => true,
-#  }
+# Add a class for each check instance:
+#
+# class { 'datadog_agent::integrations::directory':
+#   directory => '/opt/ftp_data',
+#   recursive => true,
+#   countonly => true,
+# }
+#
+# Add multiple instances in one class declaration:
+#
+# class { 'datadog_agent::integrations::directory':
+#   instances => [{
+#     'directory' => '/opt/ftp_data',
+#     'recursive' => true,
+#     'countonly' => true,
+#     },
+#     'directory' => '/opt/ftp_data-staging',
+#     'recursive' => true,
+#     'countonly' => true,
+#     },
+#   ]
+# }
+
 class datadog_agent::integrations::directory (
-  $directory     = undef,
-  $tag_name      = '',
-  $pattern       = '*',
-  $recursive     = false
+  Array[Struct[{
```

Comment:
This isn't supported in older puppets, so we're going to have to something suboptimal like we do with `haproxy` or `postgres`. I know, its unfortunate thay we have to prescind of a clean elegant solution like this one to accomodate for older puppets. This sort of thing should become the norm soon.